pennylane/devices/default_qubit.py

```diff
@@ -242,6 +242,8 @@ def _apply_operation(self, state, operation):
         Returns:
             array[complex]: output state
         """
+        if operation.base_name == "Identity":
+            return state
```

Comment:
Are there any other "no-ops" to include here? e.g., things like a barrier

Reply:
`Barrier` and `WireCut` don't define a matrix, so we don't have to worry about matrix operations with them.  They get expanded away to nothing prior to execution. `Identity`, on the other hand, is an explicitly supported gate, so it won't get expanded away.